src/addons/grids/ReactDataGrid.js

```diff
@@ -480,9 +482,27 @@ const ReactDataGrid = React.createClass({
     // we need to prevent default as we control grid scroll
     // otherwise it moves every time you left/right which is janky
     e.preventDefault();
-    let rowIdx = this.state.selected.rowIdx + rowDelta;
+    let rowIdx = this.state.selected.rowIdx;
     let idx = this.state.selected.idx + cellDelta;
-    this.onSelect({idx: idx, rowIdx: rowIdx});
+    let _rowDelta = rowDelta;
+    const { enableCellSelectionLoop } = this.props;
+    if (cellDelta > 0) {
```

Comment:
because the default behavior would be for it to move to first cell in next row, last cell in previous row. i think the prop name might be confusing, by enableCellSelectionLooping, i mean to make it loop around to the beginning/end of the row. Otherwise, when set to false, it would move to next/previous row rather than do nothing